### PR TITLE
Auto S3 for Heroku

### PR DESCRIPTION
**Pull request by Lewiscowles1986**
Brings AWS S3 for Heroku (no setup needed)

### CODE REVIEW

File: config/filesystems.php [hunk 1]
```diff
@@ -59,11 +59,31 @@
             'driver' => 's3',
             'key' => env('AWS_ACCESS_KEY_ID'),
             'secret' => env('AWS_SECRET_ACCESS_KEY'),
-            'region' => env('AWS_DEFAULT_REGION'),
-            'bucket' => env('AWS_BUCKET'),
-            'url' => env('AWS_URL'),
+            'region' => env('AWS_DEFAULT_REGION', 'us-east-1'),
```
Comment (aaronpk) [inline config/filesystems.php]: Shouldn't the region also be set from the URL? Or does it not provide that?

File: config/filesystems.php [hunk 2]
```diff
@@ -59,11 +59,31 @@
...
+            'url' => env('AWS_URL', get_aws_url_from_env()),
         ],
 
     ],
 
 ];
+
+function get_url_path($url) {
+    $parsed = parse_url($url);
+    return array_key_exists('path', $parsed) ? $parsed['path'] : '';
```
Comment (aaronpk) [inline config/filesystems.php]: Minor nit, this can be:

`$parsed['path'] ?? ''`

File: config/filesystems.php [hunk 3]
```diff
@@ -59,11 +59,31 @@
...
+}
+
+function get_url_without_path($url, $path=null) {
+    $path = $path ?? '';
+    return substr($url, 0, strlen($url) - strlen($path));
+}
+
+function get_aws_bucket_from_env() {
```
Comment (aaronpk) [inline config/filesystems.php]: Can you provide a comment here with an example URL that's being parsed? I'm not familiar with the strings set by the heroku addon and it'd be helpful to have that here as a reference when I'm looking at this code in the future

File: config/filesystems.php (hunk 1 above)
Comment (Lewiscowles1986) [reply config/filesystems.php]: it does. Check the lastest commit 😉

File: config/filesystems.php (hunk 2 above)
Comment (Lewiscowles1986) [reply config/filesystems.php]: make sure it's a string

File: config/filesystems.php (hunk 3 above)
Comment (Lewiscowles1986) [reply config/filesystems.php]: I gave examples in the commits, but I am now looking to extract this to a helper to use in the setup controller so we can write to config vars and not parse per-request 👍

Comment (Lewiscowles1986) [reply config/filesystems.php]: Also tests 👍

**Pull request merged**